app/controllers/jira/status/push_controller.rb

```diff
@@ -71,15 +71,16 @@ def update
         else
           flash[:alert] = 'No changes made'
         end
-        redirect_to action: 'edit', id: @push.head_commit.sha
+        redirect_to action: 'edit', id: @push.head_commit.sha, service_name: @push.service_name
       end
 
       def branch
         render 'edit'
       end
 
       def summary
-        @push = Branch.where(name: 'master').first!.pushes.last
+        # Only used for web
+        @push = Branch.where(name: 'master').first!.pushes.for_ancestor("web").last
```

Comment:
Ah I see why there's only one `summary`. How much work would it be to extend this to have a summary page per service name?

Reply:
updated to have service specific summary pages